packages/realm/bindgen/src/templates/node-wrapper.ts

```diff
@@ -122,6 +119,29 @@ export function generate({ spec: rawSpec, file }: TemplateContext): void {
     }
   `);
 
+  both(`
+    // Wrapped types
+    export class Float {
+      constructor(value) {
+        this.value = value;
+      }
+      valueOf() {
+        return this.value;
+      }
+    }
+  `);
+
+  both("// Enums");
+  for (const e of spec.enums) {
+    // This pattern is copied from the TS compiler
+    both(`export var ${e.jsName};`);
+    both(`(function (${e.jsName}) {`);
+    both(
+      ...e.enumerators.map(({ jsName, value }) => `${e.jsName}[${e.jsName}["${jsName}"] = ${value}] = "${jsName}";`),
+    );
+    both(`})(${e.jsName} || (${e.jsName} = {}));`);
+  }
```

Comment:
Probably the most controversial change, but not a large price to pay to avoid relying on code replacements from the Rollup config.

Reply:
const enums don't generate _any_ js code: https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM72MgrgWzgMQgjgG8BYAKDmrgDMiAaOAIwEMom2AvSgXyA. It only adds declarations to the .d.ts file.

Reply:
I believe `const enum` is only supposed to be used in non-declaration files.

Here's an example:

```json
{
  "name": "ts-const-enums",
  "private": true,
  "type": "module",
  "scripts": {
    "test": "tsx index.ts"
  },
  "dependencies": {
    "tsx": "^3.12.7",
    "typescript": "^5.0.4"
  }
}
```

```ts
// index.ts
import { Foo } from "./enums";
console.log(Foo.bar);
```

```ts
// enums.d.ts
export const enum Foo {
  bar = "bar"
}
```

```ts
// enums.js
// note how we provide no implementation
```

Running `npm test` yeilds:

```
/Users/kraen.hansen/Projects/ts-const-enums/index.ts:1
import { Foo } from "./enums";
         ^

SyntaxError: The requested module './enums' does not provide an export named 'Foo'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:123:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:189:5)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:527:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)
```

Even if we do provide a dummy implementation of `Foo` in `enums.js`:

```js
export const Foo = {};
```

This simply prints `undefined`.

Reply:
Maybe it behaves differently for string vs number `const enum`s? When I tested locally with this patch, everything seems to work:

```diff
diff --git a/packages/realm/bindgen/src/templates/node-wrapper.ts b/packages/realm/bindgen/src/templates/node-wrapper.ts
index c66cf638c..364d7d840 100644
--- a/packages/realm/bindgen/src/templates/node-wrapper.ts
+++ b/packages/realm/bindgen/src/templates/node-wrapper.ts
@@ -131,17 +131,6 @@ export function generate({ spec: rawSpec, file }: TemplateContext): void {
     }
   `);

-  both("// Enums");
-  for (const e of spec.enums) {
-    // This pattern is copied from the TS compiler
-    both(`export var ${e.jsName};`);
-    both(`(function (${e.jsName}) {`);
-    both(
-      ...e.enumerators.map(({ jsName, value }) => `${e.jsName}[${e.jsName}["${jsName}"] = ${value}] = "${jsName}";`),
-    );
-    both(`})(${e.jsName} || (${e.jsName} = {}));`);
-  }
-
   const injectables = [
     "Long",
     "ArrayBuffer",
diff --git a/packages/realm/bindgen/src/templates/typescript.ts b/packages/realm/bindgen/src/templates/typescript.ts
index 3210f1f5c..0091d5f34 100644
--- a/packages/realm/bindgen/src/templates/typescript.ts
+++ b/packages/realm/bindgen/src/templates/typescript.ts
@@ -203,7 +203,7 @@ export function generate({ spec: rawSpec, file }: TemplateContext): void {
   out("// Enums");
   for (const e of spec.enums) {
     // Using const enum to avoid having to emit JS backing these
-    out(`export enum ${e.jsName} {`);
+    out(`export const enum ${e.jsName} {`);
     out(...e.enumerators.map(({ jsName, value }) => `${jsName} = ${value},\n`));
     out("};");
   }
```

Of course, I've often wondered whether we should be using `enum` or `const enum` for these. `const enum` may provide some perf and size benefits, but I don't know how much that matters given the JIT in node and the bytecode compiler for hermes. But it certainly shouldn't be slower, and I don't think we've needed the features of a non-const enum yet (eg mapping numeric values to/from string names at runtime)

Reply:
When I apply this, the SDK unit test fails with:

```
✖ ERROR: TypeError: Cannot read properties of undefined (reading 'Get')
    at <anonymous> (/Users/kraen.hansen/Projects/realm-js/packages/realm/src/platform/network.ts:35:23)
    at Object.<anonymous> (/Users/kraen.hansen/Projects/realm-js/packages/realm/src/platform/network.ts:62:17)
```